lib/API/DX/Device.cpp

```diff
@@ -653,19 +732,75 @@ class DXDevice : public offloadtest::Device {
               "Failed to create committed resource (readback buffer)."))
         return Err;
 
-      // Initialize the UAV data
+      // Tile mapping setup (optional if NumTiles > 0)
+      const UINT NumTiles = getNumTiles(R.TilesMapped, ResDesc.Width);
+
+      ComPtr<ID3D12Heap> Heap; // optional, only created if NumTiles > 0
+
+      if (NumTiles > 0) {
```

Comment:
This whole block (up to the `updateTileMappings` call) looks identical to the SRV version. Can we pull out a helper function to call in both cases?

Reply:
Turns out I'll have to leave unmapped UAVs for another time, so I'll pull out a helper function in a future PR.